tests/test_Step.cc

```diff
@@ -428,16 +428,16 @@ TEST_CASE("execute(): C++ exceptions", "[Step]")
         step.set_script("throw_logic_error(); a = 42");
 
         // Like a genuine Lua exception, a C++ exception must be reported as a
-        // task::ErrorAtIndex.
-        REQUIRE_THROWS_AS(step.execute(context), ErrorAtIndex);
+        // task::Error.
+        REQUIRE_THROWS_AS(step.execute(context), Error);
         REQUIRE(std::get<VarInteger>(context.variables["a"]) == 0);
     }
 
     SECTION("Nonstandard C++ exception thrown at script runtime")
     {
         step.set_script("throw_weird_exception(); a = 42");
 
-        REQUIRE_THROWS_AS(step.execute(context), ErrorAtIndex);
+        REQUIRE_THROWS_AS(step.execute(context), Error);
```

Comment:
I would also add new test case to check `OptionalStepIndex` (... maybe also for `test_Sequence.cc`?). I know that we have tests for `test_Executor.cc` but I would also expect it here.

Reply:
You are right, we are not checking the content of the thrown exception yet. Will add something.

Reply:
Hey, wait – why is this marked as "resolved"? I did not do anything yet...

Reply:
Okay, this lead me down a rabbit hole... I found that we still had the `Step::execute()` overload with 1 parameter that used a `StepIndex` of zero as a default, instead of a `nullopt`. I went in and unified the two `execute()` overloads into one and modified all lower-level functions to deal with `OptionalStepIndex` instead of `StepIndex`. Then I added two small tests for correct reporting of the step index in exceptions thrown by `execute()`. Please have a look.

Reply:
Looks ok 👍 !